src/Microsoft.TestPlatform.CommunicationUtilities/TestRequestSender.cs

```diff
@@ -609,7 +612,24 @@ private void OnDiscoveryMessageReceived(ITestDiscoveryEventsHandler2 discoveryEv
             OnDiscoveryAbort(discoveryEventsHandler, ex, false);
         }
     }
+    private static bool IsBlameEnabled()
+    {
+        string? runSettingsXml = RunSettingsManager.Instance.ActiveRunSettings.SettingsXml;
+        DataCollectionRunSettings? dataCollectionRunSettings = XmlRunSettingsUtilities.GetDataCollectionRunSettings(runSettingsXml);
+
+        if (dataCollectionRunSettings is null)
+            return false;
+
+        foreach (var item in dataCollectionRunSettings.DataCollectorSettingsList)
+        {
+            if (item.FriendlyName?.ToLowerInvariant() == "blame" && item.IsEnabled)
```

Comment:
Change the order of the && to make it fail early on isEnabled == false, and check the equality by .Equals with OrdinalCaseInsensitive (or do we have an extension method for that already)?